### PR TITLE
Add `class` template variable back.

### LINKED ISSUE #17008: Undefined variable $class in missing_controller.php
### Description

My tests fail after the update from 4.4.10 to 4.4.11

warning: 2 :: Undefined variable $class on line 24 of /app/vendor/cakephp/cakephp/templates/Error/missing_controller.php
Stack Trace:

include /app/vendor/cakephp/cakephp/templates/Error/missing_controller.php, line 24
Cake\View\View::_evaluate() /app/vendor/cakephp/cakephp/src/View/View.php, line 1220
Cake\View\View::_render() /app/vendor/cakephp/cakephp/src/View/View.php, line 1177
Cake\View\View::render() /app/vendor/cakephp/cakephp/src/View/View.php, line 808
Cake\Controller\Controller::render() /app/vendor/cakephp/cakephp/src/Controller/Controller.php, line 773

Warning can be reproduced on calling a non-existing route.

After some digging I found out that the variable name was changed in this commit
https://github.com/cakephp/cakephp/commit/4d0992832b784b043d3b3ae8be1e277ee8df05e2
But in https://github.com/cakephp/cakephp/blob/4.x/templates/Error/missing_controller.php the variable $class is still expected.

I can make a PR if you want.

### CakePHP Version

4.4.11

### PHP Version

8.2

### PR DESCRIPTION
**Pull request by markstory**
Fixes #17008

### CODE REVIEW

**Comment by ADmad** [comment]
Shouldn't we change the variable name in the template instead?

**Comment by othercorey** [comment]
Do we support users extending that template?

**Comment by ADmad** [comment]
I guess theoretically one could override these dev templates in their app but I really don't think we should be supporting/accounting for that.

**Comment by ADmad** [comment]
That's why they are under the core repo unlike the `error400` and `error500` templates in the app skeleton which are meant to be modified by users.

**Comment by ADmad** [comment]
I have made an alternate PR #17010.

**Comment by markstory** [comment]
> Do we support users extending that template?

We do which is why I thought the simplest solution was to add the variable back in. I'm fine with proceeding with ADmad's change though. I doubt many applications have changed the templates for missingcontroller.

**Comment by dereuromark** [comment]
We could also do both? And then drop the additional class info in 4.5?

File: src/Controller/ControllerFactory.php [hunk 1]
```diff
@@ -350,6 +350,7 @@ protected function missingController(ServerRequest $request)
             'plugin' => $request->getParam('plugin'),
             'prefix' => $request->getParam('prefix'),
             '_ext' => $request->getParam('_ext'),
+            'class' => $request->getParam('controller'),
```
Comment (dereuromark) [inline src/Controller/ControllerFactory.php]: E.g.
```suggestion
            'class' => $request->getParam('controller'), // Deprecated: Will be removed in 4.5. Use `controller` instead.
```

**Pull request merged**